### PR TITLE
Add conversions between Visibility and bool

### LINKED ISSUE #14521: Conversion between `Visibility` and `bool`
It would be useful to have these in bevy:
```rust
pub trait VisibilityToBool {
    fn to_bool(self) -> bool;
}
impl VisibilityToBool for Visibility {
    fn to_bool(self) -> bool {
        self == Visibility::Visible
    }
}
pub trait BoolToVisibility {
    fn to_visibility(self) -> Visibility;
}
impl BoolToVisibility for bool {
    fn to_visibility(self) -> Visibility {
        if self {
            Visibility::Visible
        } else {
            Visibility::Hidden
        }
    }
```
I find myself using these all the time in my project!

### PR DESCRIPTION
**Pull request by Stanleeeeey**
# Objective

Fixes #14521. 

## Solution

Added to methods to the VIsibility. 
```rs
is_visible() -> Result<bool, String>
```
 and 
```rs
visbility_from_bool(bool) -> Visibility
```

## Testing

Ran 
* `cargo run -p ci -- lints`
* `cargo run -p ci -- test`
* `cargo run -p ci -- compile`
it seems to be working. 
However I got few error messages :`ERROR bevy_log: could not set global logger and tracing subscriber as they are already set. Consider disabling LogPlugin` in `cargo run -p ci -- test`, even though all test passed. I'm not sure if that's expected behaviour

Ps. I'm new to contributing, please correct me if anything is wrong

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
I've updated your PRs title to be more descriptive :) It's helpful for looking through the tracker, but they're *also* used as commit messages once PRs merge.

I've also swapped your initial message to include the magic phrase "Fixes #ISSUE_NUMBER". This will link the PR and the issue, closing the issue once the PR merges.

File: crates/bevy_render/src/view/visibility/mod.rs
Comment (alice-i-cecile) [inline crates/bevy_render/src/view/visibility/mod.rs:54]:
```suggestion
    /// - returns `Err()` if `Visibility::Inherited`
```

File: crates/bevy_render/src/view/visibility/mod.rs
Comment (alice-i-cecile) [inline crates/bevy_render/src/view/visibility/mod.rs:55]: The error type here should either be `()` or ideally a custom Zero-Sized Type that implements the `Error` trait :) `thiserror` is useful to help implement the trait. Strings require allocations, can't be matched on and there's not much information there.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
I've left a more detailed suggestion about the error type in a comment, but I also think we should just use the `TryFrom` and `From` traits here :) More idiomatic and slightly more useful to users. Add a doc comment to the implementations to clarify the exact behavior for users.

File: crates/bevy_render/src/view/visibility/mod.rs
Comment (alice-i-cecile) [inline crates/bevy_render/src/view/visibility/mod.rs:54]: We should rename this to `InheritedVisibilityToBoolConversionError`;

File: crates/bevy_render/src/view/visibility/mod.rs
Comment (alice-i-cecile) [inline crates/bevy_render/src/view/visibility/mod.rs:56]:
```suggestion
/// Implements conversion from bool to Visibility.
/// 
/// `true` corresponds to [`Visibility::Visible`], while false corresponds to [`Visibility::Hidden`].
```

File: crates/bevy_render/src/view/visibility/mod.rs
Comment (alice-i-cecile) [inline crates/bevy_render/src/view/visibility/mod.rs:67]:
```suggestion
/// Implements conversion from [`Visibility`] to [`bool`].
```

**Review by alice-i-cecile — COMMENTED** [review]
Just a couple of little nits now :)

**Comment by Stanleeeeey** [comment]
Should be all good now! Sorry about the mess in commits, tried to do it in a hurry.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
No worries, we squash on merge anyways :)

File: crates/bevy_render/src/view/visibility/mod.rs
Comment (janhohenheim) [inline crates/bevy_render/src/view/visibility/mod.rs:54]: Usually, these kinds of errors are `enum`s where the specific variant points towards the more exact reason. That way, we can add more variants in the future and increase specificity without breaking changes. 
The error message could also be improved if said variants held the actual value we tried to convert. 
E.g. a `VariantNotSupported(Visibility)` would be able to show an error like "The variant {:?} cannot be converted to a bool". As-is, a user running into this error will have to manually debug or look at the source to see the actual issue.

I won't block on this though; rest seems sensible :)

**Review by janhohenheim — APPROVED** [review]

File: crates/bevy_render/src/view/visibility/mod.rs
Comment (Stanleeeeey) [reply crates/bevy_render/src/view/visibility/mod.rs:54]: Hi there! Is that an implementation You suggested?
```rs
impl Display for Visibility{
    fn fmt(&self,f: &mut fmt::Formatter) -> fmt::Result{
        write!(f, "{}", self)
    }
}

/// Enum of errors that could occur during conversion to [`bool`]
/// `VariantNotSupported(Visibility)` Provided variant cannot be converted to [`bool`]
#[non_exhaustive]
#[derive(Error, Debug)]
pub enum VisibilityToBoolConversionError{
    #[error("The variant `{0:?}` cannot be converted to a bool")]
    VariantNotSupported(Visibility),
}
```

Comment (janhohenheim) [reply crates/bevy_render/src/view/visibility/mod.rs:54]: @Stanleeeeey basically, yeah :) 
The second doc comment on the enum is probably there by mistake and your `Display` is recursive, looks perfect otherwise ;)

**Comment by janhohenheim** [comment]
I don't think you need the `Display` implementation here. Can you try removing it? `#[error]` should already have taken care of everything.

**Comment by Stanleeeeey** [comment]
Yup the crate took care of it, I removed that implementation. Thanks for pointing it out!

**Review by janhohenheim — APPROVED** [review]
Looks perfect now :) Good job!

**Pull request merged**
